plugins/emc/emc_vnx_block_lun_perfdata

```diff
@@ -0,0 +1,391 @@
+ Clariion systems.
+ About VNX2 series, i don't know whether the plugin will be able to work with them. Maybe it would need some corrections
+ in command-line backend. The same situation is with other EMC systems, so i encourage you to try and fix the plugin. 
+ 
+=head1 CONFIGURATION
+
+=head2 Prerequisites
+
+ First of all, be sure that statistics collection is turned on. You can do this by typing:
+ navicli -h spa setstats -on
+ on your Control Station or locally through /opt/Navisphere 
+
+ Also, the plugin actively uses buggy "cdef" feature of Munin, and here we can be hit by the following bugs:
+ http://munin-monitoring.org/ticket/1017 - Here I have some workarounds in plugin, be sure that they are working.
+ http://munin-monitoring.org/ticket/1352 - 
+ Metrics in my plugin can be much longer than 15 characters, so you have to edit the following file:
+ /usr/share/perl5/Munin/Master/GraphOld.pm
+ Find get_field_name() function and change "15" to "255".
+
+=head2 Installation
+
+ The plugin uses SSH to connect to Control Stations. It's possible to use 'nasadmin' user, but it would be better
+ if you create read-only global user by Unisphere Client. The user should have only Operator role.
+ I created "operator" user but due to the fact that Control Stations already had one internal "operator" user,
+ the new one was called "operator1". So be careful.
+ 
+ On munin-node side choose a user which will be used to connect through SSH. Generally user "munin" is ok. Then,
+ execute "sudo su munin -s /bin/bash", "ssh-keygen" and "ssh-copy-id" to both Control Stations with newly created 
+ user.
+ 
+ Make a link from /usr/share/munin/plugins/emc_vnx_dm_basic_stats to /etc/munin/plugins/emc_vnx_dm_basic_stats_<NAME>,
+ where <NAME> is any arbitrary name of your storage system. The plugin will return <NAME> in its answer 
+ as "host_name" field.
+ Assume your storage system is called "VNX5300".
+ 
+ Make a configuration file at /etc/munin/plugin-conf.d/emc_vnx_block_lun_perfdata_VNX5300
+ 
+ [emc_vnx_block_lun_perfdata_VNX5300]
+ user munin							# SSH Client local user
+ env.username operator1						# Remote user with Operator role
+ env.cs_addr 192.168.1.1 192.168.1.2				# Control Stations addresses
+
+=head1 ERRATA
+
+ It counts Queue Length in not fully correct way. We take parameters totally from both SP's, but after we divide them
+ independently by load of SPA and SPB. Anyway, in most AAA / ALUA cases the formula is correct.
+
+=head1 HISTORY
+
+ 09.11.2016 - First Release
+ 26.12.2016 - Compatibility with Munin coding style
+
+=cut
+
+export LANG=C
+TARGET=$(echo "${0##*/}" | cut -d _ -f 6)
+SPALL="SPA SPB"
+NAVICLI="/nas/sbin/navicli"
+
+ssh_check() {
+        ssh -q $username@$1 "/nasmcd/sbin/getreason | grep -w slot_\`/nasmcd/sbin/t2slot\` | cut -d- -f1"
+}
+
+
+check_conf () {
+	if [ -z "$username" ]; then
+		echo "No username ('username' environment variable)!"
+		return 1
+	fi
+
+	if [ -z "$cs_addr" ]; then
```

Comment:
Since you're using bash, you should use bash conditional expressions everywhere instead of the test command. http://wiki.bash-hackers.org/syntax/ccmd/conditional_expression